src/Controllers/ApiController.php

```diff
@@ -27,4 +39,114 @@ public function index()
             ->withHeader('content-type', 'application/json')
             ->withContent(json_encode(['error' => 'Not implemented']));
     }
+
+
+    /**
+     * @return User
+     *
+     * Warning: I am no php coder. This code isnt intended for eternity but making our lives @36c3 a bit easier.
+     * Code is mostly harvested from /includes/controller/shifts_controller.php/shifts_json_export_controller()
+     */
+    protected function doApiAuth($key)
+    {
+        $user = auth()->apiUser('key');
+        if (
+            !preg_match('/^[\da-f]{32}$/', $key)
+            || !$user
+        ) {
+            throw new HttpForbidden('{"error":"Missing or invalid key"}', ['content-type' => 'application/json']);
+        }
+
+        if (!auth()->can('view_api')) {
+            throw new HttpForbidden('{"error":"Not allowed"}', ['content-type' => 'application/json']);
+        }
+
+        return $user;
+    }
+
+    /**
+     * @Route(“/api/shifts/my”)
+     */
+    public function getMyShifts(Request $request)
+    {
+        $key = $request->input('key');
+        $user = $this->doApiAuth($key);
+
+        $shifts = Shifts_by_user($user->id);
+
+        return $this->response
+            ->setStatusCode(200)
+            ->withHeader('content-type', 'application/json')
+            ->withContent(json_encode($shifts));
+    }
+
+    /**
+     * @Route(“/api/shifts/my”)
+     * doesnt work? dunno why result is empty.
+     */
+    public function getShiftsByAngelType(Request $request)
+    {
+        $key = $request->input('key');
+        $user = $this->doApiAuth($key);
+
+        // Shifts_by_angeltype uses only the id, required a AngelType though.
+        $angelTypeId = (int)$request->getAttribute('angeltypeid');
+        $angelType = AngelType($angelTypeId);
+        $shifts = Shifts_by_angeltype($angelType);
+
+        return $this->response
+            ->setStatusCode(200)
+            ->withHeader('content-type', 'application/json')
```

Comment:
Using something like `$this->createResponse([...])` prevents useless code duplication ;)